pkg/maps/fragmap/fragmap.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2020 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fragmap
+
+import (
+	"fmt"
+	"unsafe"
+
+	"github.com/cilium/cilium/common/types"
+	"github.com/cilium/cilium/pkg/bpf"
+	"github.com/cilium/cilium/pkg/datapath/linux/probes"
+	"github.com/cilium/cilium/pkg/logging"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/cilium/cilium/pkg/option"
+)
+
+var (
+	log = logging.DefaultLogger.WithField(logfields.LogSubsys, "map-events")
+)
+
+const (
+	mapName    = "cilium_ipv4_frag_datagrams"
+	// MaxEntries is the number of entries in the LRU map
+	MaxEntries = 8192
+)
+
+// FragmentKey must match 'struct ipv4_frag_id' in "bpf/lib/ipv4.h".
+// +k8s:deepcopy-gen=true
+// +k8s:deepcopy-gen:interfaces=github.com/cilium/cilium/pkg/bpf.MapKey
+type FragmentKey struct {
+	destAddr   types.IPv4 `align:"daddr"`
+	sourceAddr types.IPv4 `align:"saddr"`
+	id         uint16     `align:"id"`
+	proto      uint8      `align:"proto"`
+}
+
+// FragmentValue must match 'struct ipv4_frag_l4ports' in "bpf/lib/ipv4.h".
+// +k8s:deepcopy-gen=true
+// +k8s:deepcopy-gen:interfaces=github.com/cilium/cilium/pkg/bpf.MapValue
+type FragmentValue struct {
+	destPort   uint16 `align:"dport"`
+	sourcePort uint16 `align:"sport"`
+}
```

Comment:
You can add these to [`datapath/alignchecker/`](https://github.com/cilium/cilium/blob/master/pkg/datapath/alignchecker/alignchecker.go).

Reply:
Thanks, will do!

Reply:
While I'm on it: you'll also need to update `bpf/bpf_alignchecker.c`.